src/transformStateWithClones.js

```diff
@@ -6,8 +6,37 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrayStateConditions = [];
+  let stateNew = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties': {
+        stateNew = { ...stateNew, ...action.extraData };
+        break;
+      }
+
+      case 'removeProperties': {
+        stateNew = { ...stateNew };
```

Comment:
Creating a new copy of `stateNew` in this line is unnecessary since you are already working with a copy of the state. You can safely remove this line.